Snippets/Snippets_6/Host/EndpointStartAndStop.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace Snippets6.Host
+{
+    using System.Threading.Tasks;
+    using NServiceBus;
+
+    #region host-EndpointStartAndStop
+
+    class RunWhenEndpointStartsAndStops : IWantToRunWhenEndpointStartsAndStops
+    {
+        public async Task Start(IMessageSession session)
+        {
+            // perform startup logic
+        }
+
+        public async Task Stop(IMessageSession session)
```

Comment:
isnt IMessageSession the wrong type for endpoint shutdown? cc @danielmarbach @timbussmann